onnxruntime/core/providers/nnapi/nnapi_builtin/builders/model_builder.cc

```diff
@@ -439,45 +459,52 @@ uint32_t ModelBuilder::AddOperandFromPersistMemoryBuffer(
     auto persist_buffer = std::make_unique<Model::NNMemory>(nnapi_, name.c_str(), padded_size);
     uint8_t* dest = persist_buffer->GetDataPtr();
     memcpy(dest, buffer, size);
-    SetOperandValue(index, persist_buffer.get(), size, 0);
+    ORT_RETURN_IF_ERROR(SetOperandValue(index, persist_buffer.get(), size, 0));
     nnapi_model_->mem_persist_buffers_.push_back(std::move(persist_buffer));
   }
 
-  return index;
+  return Status::OK();
 }
 
-void ModelBuilder::AddOperations() {
-  const auto& node_indices = graph_view_.GetNodesInTopologicalOrder();
+Status ModelBuilder::AddOperations() {
+  const auto& node_indices = graph_viewer_.GetNodesInTopologicalOrder();
   for (size_t i = 0; i < node_indices.size(); i++) {
-    const auto* node(graph_view_.GetNode(node_indices[i]));
+    const auto* node(graph_viewer_.GetNode(node_indices[i]));
     if (auto* op_builder = GetOpBuilder(*node)) {
       op_builder->AddToModelBuilder(*this, *node);
     } else {
-      ORT_THROW("Node [" + node->Name() + "], type [" + node->OpType() + "] is not supported");
+      return Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT,
+                    "Node [" + node->Name() + "], type [" + node->OpType() + "] is not supported");
     }
```

Comment:
Prefer ORT_MAKE_STATUS. It will use a stringstream to construct the message.

e.g. return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Node [", node->Name(), "], type [", node->OpType(), "] is not supported");